### PR TITLE
feat: provena python client integration

### PR DESCRIPTION
**Pull request by PeterBaker0**
Replaces `mdsisclienttools` with `provenaclient` from https://github.com/provena/provena-python-client. 

This saves code duplication and simplifies the authorisation workflow.

Also

- added configuration workflow which is currently entirely opt-in, allows runtime overrides of the variables with a `.env` file
- removed the connectivity code since it was never used
- removes some unused imports

### CODE REVIEW

File: rrap_dg/config.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    provena_realm_name: str = "rrap"
+    # The keycloak client ID
+    provena_client_id: str = "automated-access"
+
+    class Config:
+        env_file = ".env"
+        env_file_encoding = "utf-8"
+
+
+def get_settings() -> Settings:
```
Comment (ConnectedSystems) [inline rrap_dg/config.py:20]: Is this function necessary?

File: rrap_dg/config.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    """
+
```
Comment (ConnectedSystems) [inline rrap_dg/config.py:22]: Errant space?

**Review by ConnectedSystems — APPROVED** [review]
Happy to merge in if I can get responses to two minor comments.

File: rrap_dg/config.py (hunk 1 above)
Comment (PeterBaker0) [reply rrap_dg/config.py:20]: Not necessary but I like having that there to block the rest of the application from directly depending on a pydantic model which could change interfaces.

File: rrap_dg/config.py (hunk 2 above)
Comment (PeterBaker0) [reply rrap_dg/config.py:22]: Will clean up

File: rrap_dg/config.py (hunk 1 above)
Comment (ConnectedSystems) [reply rrap_dg/config.py:20]: Ah okay, that makes sense

**Comment by ConnectedSystems** [comment]
Thanks @PeterBaker0

**Pull request merged**